ekncontent/ekncontent/eknc-utils.c

```diff
@@ -196,6 +196,30 @@ eknc_get_ekn_version (const gchar *app_id,
   return g_strdup (stripped);
 }
 
+/**
+ * eknc_get_current_language:
+ *
+ * Gets the xapian friendly version of the current system language, or NULL if
+ * none set.
+ *
+ * Returns: the language
+ */
+const gchar *
+eknc_get_current_language (void)
+{
+  const gchar * const * langs = g_get_language_names ();
+
+  guint length = g_strv_length ((gchar **)langs);
+  for (guint i = 0; i < length; i++)
```

Comment:
This has a difference in behaviour with the JS version: it will return `"C"` if it gets to the end of the list. I don't think that's what you want, because C isn't a valid language for Xapian.